kubetools/kubernetes/api.py

```diff
@@ -41,9 +41,29 @@ def _get_k8s_apps_api(env):
     return client.AppsV1Api(api_client=api_client)
 
 
-def _get_k8s_batch_api(env):
+def check_if_cronjob_compatible(env):
+    settings = get_settings()
+
+    api_core = _get_k8s_core_api(env)
+    list_node = api_core.list_node().items
+    first_node_info = list_node[0].status.node_info
+    kubelet_version = first_node_info.kubelet_version.replace('v', '').split(".")
+    major_version = int(kubelet_version[0])
+    minor_version = int(kubelet_version[1])
+
+    if major_version > 1 or (major_version == 1 and minor_version >= 21):
+        settings.IS_CRONJOB_COMPATIBLE = True
```

Comment:
Please remove `IS_CRONJOB_COMPATIBLE`, this is not a setting. Instead, use the `check_if_cronjob_compatible` function (see [this suggestion](https://github.com/EDITD/kubetools/pull/131#discussion_r908344658))

Reply:
Here https://github.com/EDITD/kubetools/pull/131/commits/151a47065791870b477884869c2306d1589db324

Reply:
The change will have to be wider than this from what I can see.

A core problem shows up is that the configuration generated now depends on the version of the target k8s cluster, but we  don't always have a target server, for example in tests, and also in the `config` command.

1) We should make sure that we don't fail if there is no Cronjob definition in the config file. This is important for backwards compatibility
2) For unit tests, the functionality of using the `env` variable to figure out the version of the target cluster could be mocked. This also means that additional unit tests need to be added to check that functionality.
3) For the `config` command, we need to figure out what solution we want to adopt. One solution would be to add a flag to the command line to force the user to pass the target environment. This goes a little against the idea of the commad though which is to generate k8s config files independently from a cluster. Another solution would be to add something to the configuration file that we can use to determine the version of the batch API to use. I have a preference on this one, with a new entry under `cronjobs` specifying the version of the k8s API to use (`batch-api-version` ?). We could even make it default to the most recent version of the API to make it more transparent to the user.

Reply:
> We could even make it default to the most recent version of the API to make it more transparent to the user

Let's suppose that the user use k8s version < 1.21+ and doesn't provide the API version, if by default we put the new batch API version, the code will crash with an unreadable error message, I think it's better to force it, no ?

Reply:
Let's suppose that the user uses k8s version > 1.24 and doesn't provide the API version instead, this is much more likely to happen.

Reply:
Here is a fix for the tests: https://github.com/EDITD/kubetools/pull/131/commits/8839b6055f2594a8df839ca96a4a8e7ced993814

The build is failing because of `flake8` command not because of the tests